src/systems/vehicle_damage/VehicleDamage.cc

```diff
@@ -0,0 +1,192 @@
+/*
+ * Copyright (C) 2020 Open Source Robotics Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+#include <google/protobuf/message.h>
+#include <ignition/msgs/double.pb.h>
+
+#include <ignition/gazebo/components/AngularVelocity.hh>
+#include <ignition/gazebo/components/Link.hh>
+#include <ignition/gazebo/components/LinearVelocity.hh>
+#include <ignition/gazebo/components/Name.hh>
+#include <ignition/gazebo/components/Pose.hh>
+#include <ignition/gazebo/components/World.hh>
+#include <ignition/gazebo/components/Inertial.hh>
+#include <ignition/gazebo/EntityComponentManager.hh>
+#include <ignition/gazebo/Link.hh>
+#include <ignition/gazebo/Model.hh>
+#include <ignition/gazebo/Util.hh>
+
+#include <ignition/plugin/Register.hh>
+
+#include <ignition/transport/Node.hh>
+
+#include "VehicleDamage.hh"
+
+using namespace ignition;
+using namespace gazebo;
+using namespace systems;
+
+class ignition::gazebo::systems::VehicleDamagePrivate
+{
+  private: Entity baseLinkEntity;
+  private: std::string modelName;
+  private: double prevKineticEnergy {0.0};
+  private: double keThreshold;
+  /// \brief Ignition communication publisher.
+  private: transport::Node::Publisher pub;
+
+  public: Model model;
+
+  public: void Load(EntityComponentManager &_ecm,
+                  const std::shared_ptr<const sdf::Element> &_sdf)
+  {
```

Comment:
We usually keep the declaration here, and implement the function separately below, so it's a bit easier to see the documentation for all members.

Also, since these functions are only called once, I'm wondering if they're necessary, or if the implementation could just be left within `VehicleDamage::Configure` and `VehicleDamage::PostUpdate`.

Reply:
Fixed in 7c84f8ed87f1f3997138d1859c9f95a3d63078e8